honeybee_grasshopper_core/src/HB Apertures by Ratio.py

```diff
@@ -0,0 +1,209 @@
+            If an array of text or material objects are input here, different
+            materials will be assigned based on cardinal direction, starting
+            with north and moving clockwise.
+        _run: Set to True to run the component.
+    
+    Returns:
+        report: Reports, errors, warnings, etc.
+        hb_obj: The input Honeybee Face or Room with Apertures generated from
+            the input parameters.
+"""
+
+ghenv.Component.Name = "HB Apertures by Ratio"
+ghenv.Component.NickName = 'AperturesByRatio'
+ghenv.Component.Message = '0.1.0\nOCT_28_2019'
+ghenv.Component.Category = "HoneybeeCore"
+ghenv.Component.SubCategory = '0 :: Create'
+ghenv.Component.AdditionalHelpFromDocStrings = "4"
+
+try:  # import the core honeybee dependencies
+    from honeybee.boundarycondition import Outdoors
+    from honeybee.facetype import Wall
+    from honeybee.room import Room
+    from honeybee.face import Face
+    from ladybug_rhino.config import tolerance, conversion_factor
+    from ladybug_rhino.grasshopper import all_required_inputs
+except ImportError as e:
+    raise ImportError('\nFailed to import honeybee:\n\t{}'.format(e))
+
+try:  # import the honeybee-energy dependencies
+    from honeybee_energy.lib.constructions import window_construction_by_name
+    from honeybee_energy.construction.window import WindowConstruction
+except ImportError:
+    pass  # honeybee-energy is not installed and ep_constr_ will not be avaialble
+
+
+def can_host_apeture(face):
+    """Test if a face is intended to host apertures (according to this component)."""
+    return isinstance(face.boundary_condition, Outdoors) and \
+        isinstance(face.type, Wall)
+
+
+def assign_apertures(face, sub, rat, hgt, sil, hor, vert, op, ep, rad):
+    """Assign apertures to a Face based on a set of inputs."""
+    if sub:
+        face.apertures_by_ratio_rectangle(rat, hgt, sil, hor, vert, tolerance)
+    else:
+        face.apertures_by_ratio(rat, tolerance)
+    
+    # try to assign the operable property
+    if op:
+        for ap in face.apertures:
+            ap.is_operable = op
+    
+    # try to assign the energyplus construction
+    if ep is not None:
+        for ap in face.apertures:
+            ap.properties.energy.construction = ep
+
+
+def orient_index(face_orient, angles):
+    """Get the index to be used for a given face orientation in a list of angles."""
+    for i, ang in enumerate(angles):
+        if face_orient < ang:
+            return i
+    return 0
+
+
+def inputs_by_index(count, all_inputs):
+    """Get all of the inputs of a certain index from a list of all_inputs."""
+    return (inp[count] for inp in all_inputs)
+
+
+if all_required_inputs(ghenv.Component) and _run is True:
+    # duplicate the initial objects
+    hb_obj = [obj.duplicate() for obj in _hb_obj]
+    
+    # set defaults for any blank inputs
+    conversion = conversion_factor()
```

Comment:
I cold change it but, in that case it should be called `convert_to_meters` (not `convert_from_meters`) because this factor is supposed to be multiplied by any Rhino unit to convert it to meters.

The reason why we are dividing in this case is because we are starting with a unit that is supposed to be meters (like 2 meters for the default window height) and we are converting it from meters to Rhino model units so that it can be used to generate the window geometry in Rhino model units.

Let me know if you still feel that the name change is helpful now that you can see it didn't necessarily help with your understanding of the code here.

Reply:
I like convert_to_meters much better.

Reply:
Done!